packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -170,6 +170,75 @@ export default class RoutesHandler {
         }
     };
 
+    /**
+     * Handler for retrieving existing controller extension data from the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetControllerExtensionData = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const params = req.params as { controllerName: string };
+            const controllerName = sanitize(params.controllerName);
+            const codeExtFiles = await this.readAllFilesByGlob('/**/changes/*_codeExt.change');
+
+            let controllerExists = false;
+            let controllerPath = '';
+            let controllerPathFromRoot = '';
+
+            const project = this.util.getProject();
+            const sourcePath = project.getSourcePath();
+            const rootPath = project.getRootPath();
+            const projectName = rootPath.split(/[\\/]/).slice(-1)[0];
+
+            const getPath = (projectPath: string, fileName: string) =>
+                path.join(projectPath, FolderNames.Changes, FolderNames.Coding, fileName).replace(/[\\/]/g, path.sep);
+
+            for (const file of codeExtFiles) {
+                const fileStr = await file.getString();
+                const change = JSON.parse(fileStr) as {
+                    selector: {
+                        controllerName: string;
+                    };
+                    content: {
+                        codeRef: string;
+                    };
+                };
```

Comment:
I would propose to extend https://github.com/SAP/open-ux-tools/blob/main/packages/adp-tooling/src/types.ts#L54 and use it here 😉

Reply:
Fixed in: https://github.com/SAP/open-ux-tools/pull/1408/commits/43a23c11efe5bf1c3afb14575909c52931d8f0d1

Reply:
Thx